vizro-core/src/vizro/models/_components/form/_form_utils.py

```diff
@@ -72,9 +72,29 @@ def validate_slider_value(cls, value, values):
     if value is None:
         return value
 
-    if (values["min"] is not None and not value >= values["min"]) or (
-        values["max"] is not None and not value <= values["max"]
+    lvalue, hvalue = (value[0], value[1]) if isinstance(value, list) else (value, value)
+
+    if (values["min"] is not None and not lvalue >= values["min"]) or (
+        values["max"] is not None and not hvalue <= values["max"]
     ):
         raise ValueError("Please provide a valid value between the min and max value.")
 
     return value
+
+
+def validate_step(cls, step, values):
+    """Reusable validator for the "step" argument for sliders."""
+    if step is None:
+        return step
+    min_value = values["min"]
+    max_value = values["max"]
+
+    if max_value is not None and step > (max_value - min_value):
+        raise ValueError(
+            "The step value of the slider must be less than or equal to the difference between max and min."
+        )
```

Comment:
I would make that consitent with the rest: 
```suggestion
    if values["max"] is not None and step > (values["max"] - values["min"]):
        raise ValueError(
            "The step value of the slider must be less than or equal to the difference between max and min."
        )
```